src/test/scripts/generate-circom-lib.js

```diff
@@ -0,0 +1,28 @@
+const fs = require('fs')
+const { poseidon_gencontract } = require('circomlibjs')
+
+function main() {
```

Comment:
What does this script do actually? Create a custom Posseidon implementation smart contract as raw EVM bytecode?

Reply:
Yeah, apparently circom doesn't provide a Solidity implementation of the Posseidon library, and instead expects you to generate one at runtime, depending on the number of inputs you plan on using.

To make this system work with Forge, we first compile the interface included in Semaphore, then overwrite the cache file with the generated bytecode. Not the best solution (need to open an issue in Forge to see if there's a better way), but best I could come up with.

Reply:
Would be great if you could add 👆as a comment.